front/pages/api/w/[wId]/assistant/agent_configurations/index.ts

```diff
@@ -286,30 +267,18 @@ export async function createOrUpgradeAgentConfiguration({
   auth,
   assistant,
   agentConfigurationId,
-  legacySingleActionMode,
 }: {
   auth: Authenticator;
   assistant: t.TypeOf<
     typeof PostOrPatchAgentConfigurationRequestBodySchema
   >["assistant"];
   agentConfigurationId?: string;
-  legacySingleActionMode: boolean;
-} & ( // Enforce that maxToolsUsePerRun is only (and always) defined in multi-actions mode.
-  | {
-      legacySingleActionMode: true;
-      assistant: { maxToolsUsePerRun?: undefined };
-    }
-  | { legacySingleActionMode: false; assistant: { maxToolsUsePerRun: number } }
-)): Promise<Result<AgentConfigurationType, Error>> {
+}): Promise<Result<AgentConfigurationType, Error>> {
   const { actions } = assistant;
 
-  if (legacySingleActionMode && actions.length > 1) {
-    throw new Error("Only one action is supported in legacy mode.");
-  }
-
   const maxToolsUsePerRun = assistant.maxToolsUsePerRun ?? actions.length;
 
-  if (!legacySingleActionMode) {
+  if (actions.length > 1) {
```

Comment:
@spolu Following your comment: 

> One thing I was not able to uncover is where is the logic that allows saving a legacy assistant with a null description?

> We had that since we could save legacy assistants by just updating their instructions right?
> I was expecting it to be controlled by legacySingleActionMode but you're removing it in backend agent configuration update but I don't see any code that accepts the empty description case?

This is because this part of the code did not change, this is the type used for validation: 
```
const multiActionsCommonFields = {
  name: t.union([t.string, t.null]),
  description: t.union([t.string, t.null]),
};
```

Yet here I could had a check on the description so we don't force name only if we're multi-action. WDYT?

Reply:
I guess this is fine. Maybe worth a comment to explain that `actions.length > 1`

Reply:
Done! 🙏🏻